app/controllers/miq_task_controller.rb

```diff
@@ -120,8 +120,10 @@ def delete_tasks
   def delete_all_tasks
     assert_privileges("miq_task_deleteall")
     task_ids = []
-    session[:view].table.data.each do |rec|
-      task_ids.push(rec["id"])
+    if session[:view].table.empty? == false
+      session[:view].table.data.each do |rec|
```

Comment:
Alternatively, just check for nils directly.

```suggestion
    session[:view].table&.data&.each do |rec|
```

Reply:
@Fryguy This method still results in the same error but your first suggestion works. I will make that change.

Reply:
I think this is because it is not nil but rather an empty hash instead.

Reply:
I think I had the wrong picture included in the pr description. I updated it to show the correct issue with the empty hash.

Reply:
Ok makes sense...the present? will handle both nil and empty